dm/worker/server.go

```diff
@@ -177,6 +184,41 @@ func (s *Server) Start() error {
 	return terror.ErrWorkerStartService.Delegate(err)
 }
 
+func (s *Server) syncMasterEndpoints(ctx context.Context) {
+	lastClientUrls := []string{}
+	clientURLs := []string{}
+
+	updateF := func() {
+		clientURLs = clientURLs[:0]
+		resp, err := s.etcdClient.MemberList(ctx)
+		if err != nil {
+			log.L().Error("can't get etcd member list", zap.Error(err))
+			return
+		}
+
+		for _, m := range resp.Members {
+			clientURLs = append(clientURLs, m.GetClientURLs()...)
+		}
+		if utils.NonRepeatStringsEqual(clientURLs, lastClientUrls) {
+			log.L().Debug("etcd member list doesn't change", zap.Strings("client URLs", clientURLs))
+			return
+		}
+		log.L().Debug("will sync endpoints to", zap.Strings("client URLs", clientURLs))
+		s.etcdClient.SetEndpoints(clientURLs...)
+		lastClientUrls = clientURLs
```

Comment:
if we modify `clientURLs` later, then `lastClientUrls` may be modified to the same content too. https://play.golang.org/p/sPmSP63gv6F